### PR TITLE
cmd/trace-agent: Add vector override for traces

### PR DESCRIPTION
**Pull request by prognant**
### What does this PR do?

Simple PR to add `vector.traces.url` and `vector.traces.enabled` (like metrics & logs).

### Motivation

Consistency with other `vector.(logs|metrics).(enabled|url)` config keys.

### Additional Notes

N/A

### Possible Drawbacks / Trade-offs

N/A

### Describe how to test/QA your changes

With `vector.traces.*` disabled ensure that there is no unforeseen behaviour changes.
And with the newer option set traces should be sent to the url set in `vector.traces.url` (it is really just an alias of `apm_config.apm_dd_url`).

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by ajgajg1134 — APPROVED** [review]
One question: So if someone is using vector it should replace the normal apm_dd_url that is set? (And I assume the vector pipeline will move the traces through some flow that eventually leads it somewhere like apm_dd_url?) Other than that this looks good to me!

**Review by cswatt — COMMENTED** [review]
edited release note for grammar

File: cmd/trace-agent/config/testdata/vector_override.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+api_key: api_key_test
+dd_url: https://app.datadoghq.com
+site: datadoghq.eu
+vector.traces:
```
Comment (gbbr) [inline cmd/trace-agent/config/testdata/vector_override.yaml:4]:
```suggestion
vector
  traces:
```
Shouldn't it be like this? Even if it's the same thing, I think it's more correct as an example.

File: pkg/config/apm.go [hunk 2]
```diff
@@ -16,6 +16,9 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
+// Traces type used for Vector override
```
Comment (gbbr) [inline pkg/config/apm.go:19]:
```suggestion
// Traces specifies the data type used for Vector override.
```
What is Vector? If the reader doesn't know, can you provide some insight here?

File: releasenotes/notes/vector-traces-override-bebe167f286f7a3d.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+features:
+  - |
+    Add two options under the `vector` config prefix to send traces
```
Comment (gbbr) [inline releasenotes/notes/vector-traces-override-bebe167f286f7a3d.yaml:11]:
```suggestion
    APM: Add two options under the `vector` config prefix to send traces
```

**Review by gbbr — CHANGES_REQUESTED** [review]

File: cmd/trace-agent/config/config.go [hunk 4]
```diff
@@ -138,7 +138,11 @@ func applyDatadogConfig(c *config.AgentConfig) error {
 		c.StatsdPort = coreconfig.Datadog.GetInt("dogstatsd_port")
 	}
 
-	c.Endpoints[0].Host = coreconfig.GetMainEndpoint(apiEndpointPrefix, "apm_config.apm_dd_url")
+	if coreconfig.Datadog.GetBool("vector.traces.enabled") && coreconfig.Datadog.IsSet("vector.traces.url") && len(coreconfig.Datadog.GetString("vector.traces.url")) > 0 {
```
Comment (gbbr) [inline cmd/trace-agent/config/config.go:141]: Shouldn't the user know if `vector.traces.url` happens to be empty? For that reason, I suggest:
```suggestion
	if coreconfig.Datadog.GetBool("vector.traces.enabled") {
	    if host := coreconfig.Datadog.GetString("vector.traces.url"); host == "" {
	        log.Error("vector.traces.enabled but vector.traces.url is empty")
	    } else {
	        c.Endpoints[0].Host = host
	    }
```
Also, you're using `GetBool` straight up, without `IsSet`, so let's do the same for `GetString`.

**Review by gbbr — COMMENTED** [review]
One more thing...

File: cmd/trace-agent/config/config.go (hunk 4 above)
Comment (prognant) [reply cmd/trace-agent/config/config.go:141]: Interesting, I borrowed the `IsSet` before `GetString` from [here](https://github.com/DataDog/datadog-agent/blob/main/pkg/logs/config/config_keys.go#L54-L56) but your are right, I double checked and viper indeed returns "" for a non-existing key.

File: cmd/trace-agent/config/config.go [hunk 5]
```diff
@@ -138,7 +138,16 @@ func applyDatadogConfig(c *config.AgentConfig) error {
...
 	}
 
-	c.Endpoints[0].Host = coreconfig.GetMainEndpoint(apiEndpointPrefix, "apm_config.apm_dd_url")
+	if coreconfig.Datadog.GetBool("vector.traces.enabled") {
+	    if host := coreconfig.Datadog.GetString("vector.traces.url"); host == "" {
+	        log.Error("vector.traces.enabled but vector.traces.url is empty.")
+	    } else {
+	        c.Endpoints[0].Host = host
+	    }
+		c.Endpoints[0].Host = coreconfig.Datadog.GetString("vector.traces.url")
```
Comment (gbbr) [inline cmd/trace-agent/config/config.go:147]: I think you have to remove this line now.

**Review by gbbr — APPROVED** [review]
Great! Thanks for your patience!

**Review by maycmlee — APPROVED** [review]
👍 for docs.

**Pull request merged**